dlt/common/configuration/specs/aws_credentials.py

```diff
@@ -77,6 +78,16 @@ def to_object_store_rs_credentials(self) -> Dict[str, str]:
 
         return creds
 
+    def to_pyiceberg_fileio_config(self) -> Dict[str, Any]:
+        return {
+            "s3.access-key-id": self.aws_access_key_id,
+            "s3.secret-access-key": self.aws_secret_access_key,
+            "s3.session-token": self.aws_session_token,
+            "s3.region": self.region_name,
+            "s3.endpoint": self.endpoint_url,
+            "s3.connect-timeout": 300,
```

Comment:
Perhaps timeout should be configurable.

Reply:
there's no way to pass fsspec filesystem instance, right?

Reply:
ok location for tables is always `str` so the answer is no

Reply:
Right, I don't think you can pass an `fsspec` instance. `pyiceberg` uses two different filesystem backends (`fsspec` and `pyarrow`) and it seems they chose to have their own `FileIO` [config](https://py.iceberg.apache.org/configuration/#fileio) system on top of those.